api/all/src/main/java/io/opentelemetry/api/trace/TraceFlags.java

```diff
@@ -8,47 +8,104 @@
+   * @return the default (with all flag bits off) byte representation of the {@code TraceFlags}.
    */
   public static byte getDefault() {
     return DEFAULT;
   }
 
   /**
-   * Returns the byte representation of the flags with the sampling bit set to {@code 1}.
+   * Returns the byte representation of the {@code TraceFlags} with the sampling flag bit on.
    *
-   * @return the byte representation of the flags with the sampling bit set to {@code 1}.
+   * @return the byte representation of the {@code TraceFlags} with the sampling flag bit on.
    */
   public static byte getSampled() {
-    return IS_SAMPLED;
+    return SAMPLED_BIT;
+  }
+
+  /**
+   * Returns {@code true} if the sampling bit is on for this byte representation of the {@code
+   * TraceFlags}, otherwise {@code false}.
+   *
+   * @param traceFlags the byte representation of the {@code TraceFlags}.
+   * @return {@code true} if the sampling bit is on for this byte representation of the {@code *
+   *     TraceFlags}, otherwise {@code false}.
+   */
+  public static boolean isSampled(byte traceFlags) {
+    return (traceFlags & SAMPLED_BIT) != 0;
+  }
+
+  /**
+   * Returns the byte representation of the {@code TraceFlags} converted from the given hex (base16)
+   * representation.
+   *
+   * @param first the first lowercase hex (base16) character {@code TraceFlags}.
+   * @param second the second lowercase hex (base16) character {@code TraceFlags}.
+   * @return the byte representation of the {@code TraceFlags}.
+   * @throws NullPointerException if {@code traceFlagsHex} is null.
+   * @throws IllegalArgumentException if not enough characters in the {@code traceFlagsHex}.
+   */
+  public static byte fromHexCharacters(char first, char second) {
+    return BigendianEncoding.decodeByte(first, second);
   }
 
-  /** Extract the byte representation of the flags from a hex-representation. */
-  public static byte byteFromHex(CharSequence src, int srcOffset) {
-    return BigendianEncoding.byteFromBase16String(src, srcOffset);
+  /**
+   * Returns the first lowercase hex (base16) character of the {@code TraceFlags} converted from the
+   * given hex (base16) representation.
+   *
+   * @param traceFlags the byte representation of the {@code TraceFlags}.
+   * @throws IllegalArgumentException if not enough characters in the {@code dest}.
+   */
+  public static char firstHexCharacter(byte traceFlags) {
+    // TODO: Optimize this if we go with this version.
+    char[] dest = new char[2];
+    BigendianEncoding.byteToBase16String(traceFlags, dest, 0);
+    return dest[0];
+  }
+
+  /**
+   * Returns the second lowercase hex (base16) character of the {@code TraceFlags} converted from
+   * the given hex (base16) representation.
+   *
+   * @param traceFlags the byte representation of the {@code TraceFlags}.
+   * @throws IllegalArgumentException if not enough characters in the {@code dest}.
+   */
+  public static char secondHexCharacter(byte traceFlags) {
```

Comment:
Better names can be provided as `secondCharacterAsHex` or `lowPartAsHex`, similar for the `firstHexCharacter`